docs/development.md

```diff
@@ -46,8 +46,77 @@ Edit docker-compose.yml configuration file to change the default ports.
  ```
 ### Testing
 
-TODO: add
-
+We strive to have a full test coverage in Presidio, and expect every pull request to
+ include tests. 
+ 
+ In each service directory, a 'test' directory can be found. In it, both unit tests,
+  for testing single files or classes, and integration tests, for testing integration
+   between the service components, or integration with external packages. 
+   
+#### Basic conventions
+
+For tests to be consistent and predictable, we use the following basic conventions:
+
+ 1. Treat tests as production code. Keep the tests concise and readable, with descriptive namings. 
+ 2. Assert on one behavior at a time in each test.
+ 3. Test names should follow a pattern of `test_when_[condition_to_test]_then_[expected_behaviour]`.
+ For example: test_when_no_interpretability_requested_then_response_contains_no_analysis.
+ 4. Use test doubles and mocks when writing unit tests. Make less use of them when writing integration tests.
```

Comment:
I'm not familiar with `doubles`. Maybe link to some references about doubles and mocks?